lib/nexpose/vulnerability.rb

```diff
@@ -83,13 +86,22 @@ def method_missing(method, *args)
 
       # Finally the enumerations: references, tags
       if method_name == 'references'
-        @xml.xpath("./references/reference").collect{|entry| {:source => entry['source'], :text => entry.text} }
+        return @xml.xpath("./references/reference").collect{|entry| {:source => entry['source'], :text => entry.text}}
       elsif method == 'tags'
-        @xml.xpath("./tags/tag").collect(&:text)
-      else
-        # nothing found, the tag is valid but not present in this ReportItem
-        return nil
+        return @xml.xpath("./tags/tag").collect(&:text)
       end
+
+      # Handle evidence creation
+      if method_name == 'details'
+        vuln_id = @xml.attributes['id'].value
+
+        return @xml.xpath("//test[@id='#{vuln_id}']/Paragraph").
```

Comment:
there is some duplication between this code and that of line #L81
We might be better off extracting this functionality into a series of private methods that actually do some proper parsing of lists, code blocks, links, etc.